lib/rules/no-useless-non-capturing-group.ts

```diff
@@ -56,10 +70,29 @@ export default createRule("no-useless-non-capturing-group", {
             node,
             getRegexpLocation,
             fixReplaceNode,
+            getUsageOfPattern,
         }: RegExpContext): RegExpVisitor.Handlers {
+            let isIgnore: (gNode: Group) => boolean
+            if (allowTop === "always") {
+                isIgnore = isTopLevel
+            } else if (allowTop === "partial") {
+                const usageOfPattern = getUsageOfPattern()
+                if (
+                    usageOfPattern === UsageOfPattern.partial ||
+                    usageOfPattern === UsageOfPattern.mixed
+                ) {
+                    isIgnore = isTopLevel
```

Comment:
I would prefer this to be more conservative. The usage might be `UsageOfPattern.unknown`. If we don't know the usage, then we shouldn't assume that it's okay to remove the top-level group.

```suggestion
                if (usageOfPattern !== UsageOfPattern.whole) {
                    isIgnore = isTopLevel
```